clients/flash/air-client/src/org/bigbluebutton/air/screenshare/views/ScreenshareDock.as

```diff
@@ -0,0 +1,224 @@
+		private var _origVidHeight:int;
+		private var _sv:StageVideo;
+		private var _ssView:ScreenshareView;
+		private var _usingStageVideo:Boolean = false;
+		private var _usingVideo:Boolean = false;
+		private var _played:Boolean = false;
+		private var _screenshareRunningListener:Function;
+		
+		public function ScreenshareDock():void {	}
+		
+		private function onStageVideoState(event:StageVideoAvailabilityEvent):void {
+			var available:Boolean = (event.availability == StageVideoAvailability.AVAILABLE);
+			
+			//available = false; // for testing to force using Video instead of StageVideo!!!
+			
+			//trace("************ ScreenshareView: STAGE VIDEO available=" + available);
+			stage.removeEventListener(StageVideoAvailabilityEvent.STAGE_VIDEO_AVAILABILITY, onStageVideoState);
+			// Detect if StageVideo is available and decide what to do in toggleStageVideo 
+			toggleStageVideo(available);
+		}
+		
+		private function toggleStageVideo(on:Boolean):void {
+			_ns = new NetStream(_conn);
+			_ns.addEventListener(NetStatusEvent.NET_STATUS, onNetStatus);
+			_ns.addEventListener(AsyncErrorEvent.ASYNC_ERROR, onAsyncError);
+			_ns.client = this;
+			_ns.bufferTime = 0;
+			_ns.receiveVideo(true);
+			_ns.receiveAudio(false);
+			
+			//trace("**** Container w=" + this.width + ",h=" + this.height + " video w=" + _origVidWidth + ",h=" + _origVidHeight);
+			
+			var viewPort:Rectangle = positionAndSize(this.width, this.height, _origVidWidth, _origVidHeight);
+			
+			// To choose StageVideo attach the NetStream to StageVideo 
+			if (on) {				
+				_usingStageVideo = true;
+				if (_sv == null) {
+					//trace("***** Using StageVideo length=" + stage.stageVideos.length);
+					_sv = stage.stageVideos[0];
+					_sv.viewPort = viewPort;
+					_sv.addEventListener(StageVideoEvent.RENDER_STATE, stageVideoStateChange);
+					_sv.attachNetStream(_ns);
+				}
+				
+				if (_usingVideo) {
+					// If you use StageVideo, remove from the display list the 
+					// Video object to avoid covering the StageVideo object 
+					// (which is always in the background) 
+					stage.removeChild(_ssView);
+					_usingVideo = false;
+				}
+			} else {
+				// Otherwise attach it to a Video object 
+				if (_usingStageVideo) {
+					_usingStageVideo = false;
+				}
+				
+				//trace("***** Using classic Video");
+				_ssView = new ScreenshareView();
+				_ssView.x = viewPort.x;
+				_ssView.y = viewPort.y;
+				_ssView.width = viewPort.width;
+				_ssView.height = viewPort.height;
+				addElement(_ssView);
+				_ssView.display(_ns, viewPort.width, viewPort.height);
+				_usingVideo = true;
+			}
+			
+			if (!_played) {
+				_played = true;
+				//trace("***** Playing stream " + _streamId);
+				_ns.play(_streamId);
+			}
+		}
+		
+		private function stageVideoStateChange(event:StageVideoEvent):void {
+			var status:String = event.status;
+			//trace("***** stageVideoStateChange " + status + ",codec=" + event.codecInfo + ",color=" + event.colorSpace);

```

Comment:
The stageVideoStateChange can return with status being RENDER_STATUS_UNAVAILABLE. If that scenario occurs it should fallback and attach the NetStream to the Video.